paddle/phi/kernels/onednn/matmul_kernel.cc

```diff
@@ -152,6 +161,419 @@ void MatmulKernel(const Context &dev_ctx,
+  paddle::optional<memory> output_;
+  paddle::optional<inner_product_forward> mul_;
+  static constexpr bool is_int8_ = funcs::is_int8<XT>();
+};
+
+/* OT: output data type */
+template <typename XT, typename YT, typename OT>
+std::shared_ptr<MulPrimitiveFactory<XT, YT, OT>> GetPrimitiveFactory(
+    const OneDNNContext &dev_ctx,
+    const DenseTensor *input_x,
+    const DenseTensor *input_y,
+    const engine &onednn_engine) {
+  std::string key = funcs::CreateKey(dev_ctx,
+                                     phi::TransToProtoVarType(input_x->dtype()),
+                                     phi::vectorize(input_x->dims()),
+                                     phi::TransToProtoVarType(input_y->dtype()),
+                                     phi::vectorize(input_y->dims()),
+                                     dev_ctx.GetOutputsName("Out")[0]);
+  key = funcs::ExtendKeyWithThreadInfoIfNeeded(dev_ctx, key);
+
+  auto prim_creator = std::static_pointer_cast<MulPrimitiveFactory<XT, YT, OT>>(
+      dev_ctx.GetBlob(key));
+
+  if (prim_creator == nullptr) {
+    prim_creator =
+        std::make_shared<MulPrimitiveFactory<XT, YT, OT>>(onednn_engine);
+    dev_ctx.SetBlob(key, prim_creator);
+  }
+
+  return prim_creator;
+}
+
+/* XT: input x data type, YT: input y data type */
+template <typename XT, typename YT>
+inner_product_forward GetMulPrimitive(const OneDNNContext &dev_ctx,
+                                      const DenseTensor *input_x,
+                                      const DenseTensor *input_y,
+                                      DenseTensor *output,
+                                      int x_num_col_dims,
+                                      int y_num_col_dims,
+                                      const engine &onednn_engine) {
+  constexpr bool is_int8 = funcs::is_int8<XT>();
+  bool force_fp32_output =
+      dev_ctx.HasDnnAttr("force_fp32_output")
+          ? PADDLE_GET_CONST(bool, dev_ctx.GetDnnAttr("force_fp32_output"))
+          : false;
+
+  if (is_int8 && !force_fp32_output) {
+    return GetPrimitiveFactory<XT, YT, int8_t>(
+               dev_ctx, input_x, input_y, onednn_engine)
+        ->CreateMulPrimitive(
+            input_x, input_y, output, x_num_col_dims, y_num_col_dims, dev_ctx);
+
+  } else {
+    return GetPrimitiveFactory<XT, YT, float>(
+               dev_ctx, input_x, input_y, onednn_engine)
+        ->CreateMulPrimitive(
+            input_x, input_y, output, x_num_col_dims, y_num_col_dims, dev_ctx);
+  }
+}
+
+/* XT: input x data type */
+template <typename XT, typename Context>
+void MatmulWithFlattenKernelINT8(const Context &dev_ctx,
+                                 const DenseTensor &x,
+                                 const DenseTensor &y,
+                                 int x_num_col_dims,
+                                 int y_num_col_dims,
+                                 DenseTensor *out) {
+  PADDLE_ENFORCE_EQ(paddle::platform::is_cpu_place(dev_ctx.GetPlace()),
```

Comment:
I think that `dev_ctx.GetPlace().GetType() == AllocationType::CPU` would be consistent with migrating instruction: 
"platform::is_xxx_place | place.GetType() == phi::AllocationType::XXX"